From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Fri, 14 Jul 2023 16:40:18 -0700
Subject: [PATCH 9/9] Add pragmas for missing libraries and set _WIN32_WINNT to
 Windows 10

This makes GetSystemTimePreciseAsFileTime() available.

The #define value is from
https://learn.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt.
---
 include/uv/win.h | 2 +-
 src/win/util.c   | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/uv/win.h b/include/uv/win.h
index 6d0afe69e7dd4caf4c9459e548fe75cf0c51b501..613065df435d813cd517efbc138b13ee46f01f2d 100644
--- a/include/uv/win.h
+++ b/include/uv/win.h
@@ -20,7 +20,7 @@
  */
 
 #ifndef _WIN32_WINNT
-# define _WIN32_WINNT   0x0600
+# define _WIN32_WINNT   0x0A00
 #endif
 
 #if !defined(_SSIZE_T_) && !defined(_SSIZE_T_DEFINED)
diff --git a/src/win/util.c b/src/win/util.c
index 8828857db76a54d6e184c7f5714a091e92eba8ed..14295c41342b6dda16f39c5c548a6cc8269bfa0b 100644
--- a/src/win/util.c
+++ b/src/win/util.c
@@ -73,7 +73,9 @@ static char *process_title;
 static CRITICAL_SECTION process_title_lock;
 
 #pragma comment(lib, "Advapi32.lib")
+#pragma comment(lib, "Dbghelp.lib")
 #pragma comment(lib, "IPHLPAPI.lib")
+#pragma comment(lib, "Ole32.lib")
 #pragma comment(lib, "Psapi.lib")
 #pragma comment(lib, "Userenv.lib")
 #pragma comment(lib, "kernel32.lib")
